src/service_daemon.rs

```diff
@@ -698,6 +698,9 @@ fn new_socket_bind(intf: &Interface, should_loop: bool) -> Result<MioUdpSocket>
             sock.set_multicast_if_v4(ip)
                 .map_err(|e| e_fmt!("set multicast_if on addr {}: {}", ip, e))?;
 
+            sock.set_multicast_ttl_v4(255)
+                .map_err(|e| e_fmt!("set set_multicast_ttl_v4 on addr {}: {}", ip, e))?;
+
```

Comment:
There is one thing I didn't realize earlier: the section 11 in RFC 6762 only mentions "Multicast DNS responses".  What about queries?  

The TTL set on the socket would apply for all packets going out, including responses and queries. I'm wondering if TTL 255 has any side effects for the queries (its default TTL is 1).

Reply:
I think we can set the IP TTL in this function instead: 
https://github.com/keepsimple1/mdns-sd/blob/22a2688d6a15c7d0d5c535c7f1c76e891abd3de1/src/service_daemon.rs#L3258

where we can check `qtype`. If it's a response, set IP TTL to 255, otherwise set it to 1. 

And since this is a `Should` in the RFC, I think we can handle set TTL error gracefully, i.e. log a debug message but not to fail `send_dns_outgoing`.

Reply:
Using Wireshark, I confirmed that all mDNS traffic on a local network is supposed to set the TTL/hop limit to 255. Wireshark even notes that
 the TTL set by my router is incorrectly `1` and not `255` like the RFC says.

<img width="1085" alt="Screenshot 2025-06-16 at 18 57 04" src="https://github.com/user-attachments/assets/ba0965ad-1370-463b-83d9-7df5743b6d3e" />

Reply:
Setting the multicast TTL/hops is the most important part. To keep this PR simple, I won't change unicast TTL.

The way it's currently being set in this PR looks right to me:

- Affects all outbound multicast
- Wireshark notes that the TTL for a multicast query should also be 255

Reply:
I am okay with the the change although RFC 6762 does not specify TTL for query packets.  I don't any obvious problems per my knowledge.

Regarding unicast, note that RFC does mention it as well in [section 11](https://datatracker.ietf.org/doc/html/rfc6762#section-11): 
```
All Multicast DNS responses (including responses sent via unicast)
   SHOULD be sent with IP TTL set to 255.   
```

Currently we are good with multicast only as we don't support unicast yet.